assembly/src/parsers/nodes.rs

```diff
@@ -503,6 +517,19 @@ impl fmt::Display for Instruction {
     }
 }
 
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Builds a string from input vector to display push operation
+fn display_push_vec<T: fmt::Display>(f: &mut fmt::Formatter<'_>, values: &Vec<T>) -> fmt::Result {
+    let mut values_string = String::from("push");
+    for elem in values {
+        values_string.push('.');
+        values_string.push_str(&elem.to_string());
+    }
+    write!(f, "{values_string}")
```

Comment:
There is no need to create a separate string here. You can just write strings directly into `f` multiple times.